src/components/RoadDamage/index.tsx

```diff
@@ -0,0 +1,261 @@
+    if (roadDamageList.find((row) => row.job.jobId === job.jobId)) {
+      toast.info('This compute job result already is part of the map view.')
+      return
+    }
+
+    const dataForSameInputExists =
+      roadDamageList.filter(
+        (row) =>
+          job.inputDID.filter((did) => row.job.inputDID.includes(did))
+            .length === job.inputDID.length
+      ).length > 0
+
+    if (dataForSameInputExists)
+      if (
+        !confirm(
+          'Compute job results for a job with the same dataset inputs already exists. Add anyways?'
+        )
+      )
+        return
+
+    try {
+      const datasetDDO = await getAsset(job.inputDID[0], newCancelToken())
+
+      const signerToUse =
+        job.owner.toLowerCase() === autoWallet?.address.toLowerCase()
+          ? autoWallet
+          : signer
+
+      const jobResult = await ProviderInstance.getComputeResultUrl(
+        datasetDDO.services[0].serviceEndpoint,
+        signerToUse,
+        job.jobId,
+        job.results.findIndex((result) => result.filename === resultFileName)
+      )
+
+      const binary = await getResultBinaryData(jobResult)
+      const resultData = await transformBinaryToRoadDamageResult(binary)
+
+      if (!resultData) return
+
+      const newuseCaseData: RoadDamageUseCaseData = {
+        job,
+        result: resultData
+      }
+
+      await createOrUpdateRoadDamage(newuseCaseData)
+      toast.success('Added a new compute result')
+    } catch (error) {
+      LoggerInstance.error(error)
+      toast.error('Could not add compute result')
+    }
+  }
+
+  const deleteJobResultFromDB = async (job: ComputeJobMetaData) => {
+    if (!confirm(`Are you sure you want to delete the result from map view?`))
+      return
+
+    const rowToDelete = roadDamageList.find(
+      (row) => row.job.jobId === job.jobId
+    )
+    if (!rowToDelete) return
+
+    await deleteRoadDamage(rowToDelete.id)
+    toast.success(`Removed compute job result from map view.`)
+  }
+
+  const clearData = async () => {
+    if (mapData.length < 1) return
+    if (!confirm('All data will be removed from your cache. Proceed?')) return
+
+    await clearRoadDamages()
+    toast.success('Road Damage data was cleared.')
+  }
+
+  const getCustomActionsPerComputeJob = (job: ComputeJobMetaData) => {
+    const addAction = {
+      label: 'Add',
+      onClick: () => {
+        addComputeResultToUseCaseDB(job)
+      }
+    }
+    const deleteAction = {
+      label: 'Remove',
+      onClick: () => {
+        deleteJobResultFromDB(job)
+      }
+    }
+    const colorLeged = {
+      label: (
+        <span
+          className={styles.legend}
+          style={{ backgroundColor: getMapColor(job.inputDID) }}
+        />
+      ),
+      onClick: () => {
+        if (scrollToMapRef?.current) scrollToMapRef.current.scrollIntoView()
+      }
+    }
+
+    const viewContainsResult = roadDamageList.find(
+      (row) => row.job.jobId === job.jobId
```

Comment:
question: Can we be sure that job is never undefined?

Reply:
see above